src/proto/colors.rs

```diff
@@ -0,0 +1,45 @@
+//! An extension trait that provides the ability to strip IRC colors from a string
+use regex::Regex;
+
+/// An extension trait giving strings a function to strip IRC colors
+pub trait FormattedString {
```

Comment:
Given that this is an extension trait for all strings, I'd prefer for this to be named `FormattedStringExt` to reflect that. I think having `Ext` in the name is common, but if nothing else, it's good for consistency with `ChannelExt` here.